mlir/lib/Dialect/Bufferization/Transforms/EmptyTensorElimination.cpp

```diff
@@ -129,7 +148,7 @@ LogicalResult mlir::bufferization::eliminateEmptyTensors(
         &visitedOpOperands);
 
     for (Value v : emptyTensors) {
-      Operation *emptyTensorOp = v.getDefiningOp();
+      auto emptyTensorOp = v.getDefiningOp<tensor::EmptyOp>();
```

Comment:
I'd add an `assert(emptyTensorOp && "expected tensor.empty op")` to help with debugging in case some of this code changes in the future.

Reply:
Sure, missed it.